packages/hardhat-core/src/builtin-tasks/compile.ts

```diff
@@ -981,13 +972,11 @@ subtask(TASK_COMPILE_SOLIDITY_HANDLE_COMPILATION_JOBS_FAILURES)
       {
         compilationJobsCreationErrors,
       }: {
-        compilationJobsCreationErrors: CompilationJobsCreationErrors;
+        compilationJobsCreationErrors: CompilationJobCreationError[];
       },
       { run }
     ) => {
```

Comment:
This function is *huge*. I think we should move it to another module, but I didn't do it yet so that the PR it's easier to review.

Also, there's a lot of duplication here, especially between the "invalid imports" and the "invalid indirect imports" parts. Maybe we can abstract that, but I'm inclined to leave this as straightforward as possible.